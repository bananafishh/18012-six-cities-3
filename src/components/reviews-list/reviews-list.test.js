import React from 'react';
import renderer from 'react-test-renderer';

import ReviewsList from './reviews-list';

const reviews = [
  {
    id: 1,
    text: `Excellent location in a charming building. It’s about the size of a typical hotel room and 
    perhaps a little smaller than Airbnb places tend to be overall. Would recommend though as it’s 
    a great value and location.`,
    rating: 5,
    date: new Date(`2020-03-15`),
    user: {
      name: `Steven`,
      picture: `https://api.adorable.io/avatars/128/1`,
    }
  },
  {
    id: 2,
    text: `Apartment was very pretty and quaint! Location was excellent , very close to the main 
    sights in Porto. Abott greeted us and was very helpful, gave us some great tips.`,
    rating: 4,
    date: new Date(`2020-03-15`),
    user: {
      name: `Monica`,
      picture: `https://api.adorable.io/avatars/128/2`,
    }
  },
];

it(`Компонент «ReviewsList» рендерится корректно`, () => {
  const tree = renderer
    .create(
        <ReviewsList reviews={reviews}/>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
