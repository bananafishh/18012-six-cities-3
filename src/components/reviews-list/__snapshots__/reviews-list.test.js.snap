// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент «ReviewsList» рендерится корректно 1`] = `
<ul
  className="reviews__list"
>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviewer's avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="https://api.adorable.io/avatars/128/1"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Steven
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        Excellent location in a charming building. It’s about the size of a typical hotel room and 
    perhaps a little smaller than Airbnb places tend to be overall. Would recommend though as it’s 
    a great value and location.
      </p>
      <time
        className="reviews__time"
        dateTime="2020-03-15"
      >
        March 15, 2020
      </time>
    </div>
  </li>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviewer's avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="https://api.adorable.io/avatars/128/2"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Monica
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        Apartment was very pretty and quaint! Location was excellent , very close to the main 
    sights in Porto. Abott greeted us and was very helpful, gave us some great tips.
      </p>
      <time
        className="reviews__time"
        dateTime="2020-03-15"
      >
        March 15, 2020
      </time>
    </div>
  </li>
</ul>
`;
